File: app/src/main/java/com/juniperphoton/myersplash/cloudservice/DownloadService.java
package com.juniperphoton.myersplash.cloudservice;

import okhttp3.ResponseBody;
import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Streaming;
import retrofit2.http.Url;
import rx.Observable;

public interface DownloadService {
    @Streaming
    @GET
    Observable<ResponseBody> downloadFileWithDynamicUrlSync(@Url String fileUrl);
}

File: app/src/main/kotlin/com/juniperphoton/myersplash/cloudservice/DownloadService.kt
package com.juniperphoton.myersplash.cloudservice

import io.reactivex.Observable
import okhttp3.ResponseBody
import retrofit2.http.GET
import retrofit2.http.Query
import retrofit2.http.Streaming
import retrofit2.http.Url

interface DownloadService {
    @Streaming
    @GET
    fun downloadFileWithDynamicUrlSync(@Url fileUrl: String): Observable<ResponseBody>

    @GET
    fun reportDownload(@Url url: String, @Query("client_id") key: String): Observable<ResponseBody>
}